### PR TITLE
docs(router): Update description and example for Resolve interface

### PR DESCRIPTION
**Pull request by fmalcher**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the new behavior?

Give more context and explanation for the `Resolve` interface.
Particularly,  I wanted to see more emphasized that the router waits for async data.
I also updated the code sample so it uses the already known domain of `Hero`es.
The example no also uses a tree-shakable provider for the Resolver class.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/router/src/interfaces.ts [hunk 1]
```diff
@@ -265,66 +265,67 @@ export type CanDeactivateFn<T> =
  * @description
  *
  * Interface that classes can implement to be a data provider.
+ * A data provider class can be used with the router to resolve data during navigation.
+ * The interface defines a `resolve()` method that will be invoked when the navigation starts.
+ * The router will then wait for the data to be resolved before the route will finally be activated.
```
Comment (IgorMinar) [inline packages/router/src/interfaces.ts]:
```suggestion
 * The router will then wait for the data to be resolved before the route is finally activated.
```

**Review by IgorMinar — APPROVED** [review]
one language simplification, but otherwise lgtm. thank you!

**Comment by fmalcher** [comment]
Committed you suggestion, thanks

**Comment by kapunahelewong** [comment]
Hey, @fmalcher! :) Could you squash and rebase and tag me when you're ready?

**Comment by fmalcher** [comment]
@kapunahelewong Done 😇

**Comment by kapunahelewong** [comment]
Caretaker, this is a docs only change. Please ping me if I can help. Thank you!

**Pull request closed without merging**